From 897a5ef8a5df661f4cd6b87acd93a0e80dad27df Mon Sep 17 00:00:00 2001
From: Eskil Abrahamsen Blomfeldt <eskil.abrahamsen-blomfeldt@qt.io>
Date: Tue, 9 Sep 2025 14:18:00 +0200
Subject: [PATCH 017/553] wayland: Fix cross-compiling on Windows

On Windows, the single quote does not work, causing broken output
from qtwaylandscanner when it was used to cross-compile for
embedded Linux.

In the cmake macros we were using double quotes in some places and
single quotes in others. This makes it consistent and also work
on Windows.

Fixes: QTBUG-135354
Change-Id: Idb1e088168c1e1fba98775b733a54e0870d20d6c
Reviewed-by: David Edmundson <davidedmundson@kde.org>
(cherry picked from commit bd92a1d426b3a8b7ece1c23a12545cd355196732)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/tools/qtwaylandscanner/Qt6WaylandClientMacros.cmake     | 4 ++--
 src/tools/qtwaylandscanner/Qt6WaylandCompositorMacros.cmake | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/tools/qtwaylandscanner/Qt6WaylandClientMacros.cmake b/src/tools/qtwaylandscanner/Qt6WaylandClientMacros.cmake
index b4266c5581e..4f464e6f70e 100644
--- a/src/tools/qtwaylandscanner/Qt6WaylandClientMacros.cmake
+++ b/src/tools/qtwaylandscanner/Qt6WaylandClientMacros.cmake
@@ -94,8 +94,8 @@ function(qt6_generate_wayland_protocol_client_sources target)
             COMMAND Qt6::qtwaylandscanner client-code
                 "${protocol_file}"
                 --build-macro=${build_macro}
-                --header-path='${wayland_include_dir}'
-                --add-include='${qtwaylandscanner_code_include}'
+                --header-path="${wayland_include_dir}"
+                --add-include="${qtwaylandscanner_code_include}"
                 > "${qtwaylandscanner_code_output}"
             DEPENDS ${protocol_file} Qt6::qtwaylandscanner
         )
diff --git a/src/tools/qtwaylandscanner/Qt6WaylandCompositorMacros.cmake b/src/tools/qtwaylandscanner/Qt6WaylandCompositorMacros.cmake
index bf23bb727c0..37a4a79f45c 100644
--- a/src/tools/qtwaylandscanner/Qt6WaylandCompositorMacros.cmake
+++ b/src/tools/qtwaylandscanner/Qt6WaylandCompositorMacros.cmake
@@ -66,7 +66,7 @@ function(qt6_generate_wayland_protocol_server_sources target)
             COMMAND Qt6::qtwaylandscanner server-header
                 "${protocol_file}"
                 --build-macro=${build_macro}
-                --header-path='${wayland_include_dir}'
+                --header-path="${wayland_include_dir}"
                 > "${qtwaylandscanner_header_output}"
             DEPENDS ${protocol_file} Qt6::qtwaylandscanner
         )
@@ -76,7 +76,7 @@ function(qt6_generate_wayland_protocol_server_sources target)
             COMMAND Qt6::qtwaylandscanner server-code
                 "${protocol_file}"
                 --build-macro=${build_macro}
-                --header-path='${wayland_include_dir}'
+                --header-path="${wayland_include_dir}"
                 > "${qtwaylandscanner_code_output}"
             DEPENDS ${protocol_file} Qt6::qtwaylandscanner
         )
-- 
2.51.2

